packages/gatsby-source-contentful/src/source-nodes.js

```diff
@@ -202,45 +188,29 @@ export async function sourceNodes(
   )
   processingActivity.start()
 
-  // Array of all existing Contentful nodes
-  const existingNodes = getNodes().filter(
-    n =>
-      n.internal.owner === `gatsby-source-contentful` &&
-      (pluginConfig.get(`enableTags`)
-        ? n.internal.type !== `ContentfulTag`
-        : true)
-  )
-
   // Report existing, new and updated nodes
   const nodeCounts = {
     newEntry: 0,
     newAsset: 0,
     updatedEntry: 0,
     updatedAsset: 0,
-    existingEntry: 0,
-    existingAsset: 0,
-    deletedEntry: currentSyncData.deletedEntries.length,
-    deletedAsset: currentSyncData.deletedAssets.length,
+    deletedEntry: currentSyncData?.deletedEntries?.length || 0,
+    deletedAsset: currentSyncData?.deletedAssets?.length || 0,
   }
-  existingNodes.forEach(node => nodeCounts[`existing${node.sys.type}`]++)
-  currentSyncData.entries.forEach(entry =>
+
+  currentSyncData?.entries?.forEach(entry =>
     entry.sys.revision === 1 ? nodeCounts.newEntry++ : nodeCounts.updatedEntry++
   )
-  currentSyncData.assets.forEach(asset =>
+  currentSyncData?.assets?.forEach(asset =>
     asset.sys.revision === 1 ? nodeCounts.newAsset++ : nodeCounts.updatedAsset++
   )
 
   reporter.info(`Contentful: ${nodeCounts.newEntry} new entries`)
   reporter.info(`Contentful: ${nodeCounts.updatedEntry} updated entries`)
   reporter.info(`Contentful: ${nodeCounts.deletedEntry} deleted entries`)
-  reporter.info(
-    `Contentful: ${nodeCounts.existingEntry / locales.length} cached entries`
-  )
+  reporter.info(`Contentful: ${existingNodes.size} memory cached nodes`)
   reporter.info(`Contentful: ${nodeCounts.newAsset} new assets`)
   reporter.info(`Contentful: ${nodeCounts.updatedAsset} updated assets`)
-  reporter.info(
-    `Contentful: ${nodeCounts.existingAsset / locales.length} cached assets`
-  )
```

Comment:
Can we have something like this?


`  reporter.info(`Contentful: ${existingAssets.size} memory cached assets`)`

Reply:
@axe312ger I just pushed up a commit adding this, let me know what you think :)